lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2436,9 +2436,12 @@ size_t
 TypeSystemSwiftTypeRef::GetNumTemplateArguments(opaque_compiler_type_t type) {
   return m_swift_ast_context->GetNumTemplateArguments(ReconstructType(type));
 }
+
 CompilerType
 TypeSystemSwiftTypeRef::GetTypeForFormatters(opaque_compiler_type_t type) {
-  return m_swift_ast_context->GetTypeForFormatters(ReconstructType(type));
+  auto impl = [&]() { return ReconstructType({this, type}); };
```

Comment:
`ReconstructType` discussion: This fixes the test failures, but it seems like it's a step back. I suspect the fix to the test failures is to implement `DumpTypeValue`. But since `DumpTypeValue` is large, my current plan is to implement that separately, at which point I would remove `ReconstructType` if it's not needed. Thoughts?

Reply:
I think I'm missing some context here — why does just calling ReconstructType() without also calling GetTypeForFormatters() work at all?

In any case, tackling DumpTypeValue separately seems like a good plan.

Reply:
> why does just calling `ReconstructType()` without also calling `GetTypeForFormatters()` work at all

I'm not totally sure. Here's what I understand at this point:

The implementation of `SwiftASTContext::GetTypeForFormatters` seems to be a no-op. By manually inlining some function call indirections, it reduces to:

```swift
swift::Type qual_type(reinterpret_cast<swift::TypeBase *>(opaque_type));
return CompilerType(
      SwiftASTContext::GetSwiftASTContext(&qual_type->getASTContext()),
      qual_type.getPointer());
```

To match this, I made `TypeSystemSwiftTypeRef::GetTypeForFormatters` a no-op as well. The results give mixed messages:

1. The no-op implementation raises no assertions when validating against `SwiftASTContext`, which is good
2. However two tests fail to format a dictionary correctly, which is bad

My hunch was that by returning a `CompilerType` backed by the TypeRef type system affects subsequent calls to the returned `CompilerType`. By wrapping in `ReconstructType`, the returned `CompilerType` is instead backed by a `SwiftASTContext`, which makes the tests pass. But of course this isn't ultimately what we want. As I mentioned, I think `DumpTypeValue` might be the subsequent call that breaks the dictionary formatting.

Reply:
Ok. If it is really a noop, we could do a separate NFC change to hoist it into TypeSystemSwift? And probably only after doing the DumpTypeValue change since lifting the type into SiwftASTContext seems to be significant here.

Reply:
Yes, I expect it can be hoisted. I'll switch to `DumpTypeValue`.

Reply:
Turns out ReconstructType causes problems for other PRs, so it'll have to go. I'll come back to this soon.